site/utils/apiUtils/createDisruptionValidations.ts

```diff
@@ -0,0 +1,278 @@
+        redirectTo(res, errorPath);
+    }
+};
+
+export const validateSummary = (summary: string, errors: ErrorInfo[], errorId: string) => {
+    const summaryRequiredErr: ErrorInfo = {
+        id: errorId,
+        errorMessage: "Please enter a Summary",
+    };
+
+    if (!requireFieldCheck(summary)) {
+        errors.push(summaryRequiredErr);
+    } else {
+        const lengthErr: ErrorInfo = {
+            id: errorId,
+            errorMessage: "Please enter a Summary of less than 100 characters",
+        };
+
+        if (!validateLength(summary, 100)) {
+            errors.push(lengthErr);
+        }
+    }
+};
+
+export const validateDisruptionType = (
+    disruptionType: "planned" | "unplanned" | undefined,
+    errors: ErrorInfo[],
+    errorId: string,
+) => {
+    const distruptionTypeRequiredErr: ErrorInfo = {
+        id: errorId,
+        errorMessage: "Please choose a Type of Disruption",
+    };
+
+    if (!requireFieldCheck(disruptionType)) {
+        errors.push(distruptionTypeRequiredErr);
+    } else {
+        const invalidTypeErr: ErrorInfo = {
+            id: errorId,
+            errorMessage: "Invalid Disruption Type Selected. Please choose a valid Type of Disruption",
+        };
+
+        if (!validateDisruptionTypeValue(disruptionType, DISRUPTION_TYPES)) {
+            errors.push(invalidTypeErr);
+        }
+    }
+};
+
+export const requireFieldCheck = (field: string | undefined): boolean => {
+    let checkPassed = false;
+    if (field) {
+        //errors.push(error);
+        checkPassed = true;
+    }
+
+    return checkPassed;
+};
+
+export const validateDisruptionTypeValue = (
+    value: "planned" | "unplanned" | undefined,
+    validArray: string[],
+): boolean => {
+    let isValid = false;
+    if (value && validArray.includes(value)) {
+        isValid = true;
+    }
+
+    return isValid;
+};
+
+export const validateLength = (field: string, length: number): boolean => {
+    let isValid = false;
+    if (field?.length <= length) {
+        isValid = true;
+    }
+
+    return isValid;
+};
+
+export const validateDescription = (description: string, errors: ErrorInfo[], errorId: string) => {
+    const descriptionRequiredErr: ErrorInfo = {
+        id: errorId,
+        errorMessage: "Please enter a Description",
+    };
+
+    if (!requireFieldCheck(description)) {
+        errors.push(descriptionRequiredErr);
+    } else {
+        const error: ErrorInfo = {
+            id: errorId,
+            errorMessage: "Please enter a Description of less than 200 characters",
+        };
+        if (!validateLength(description, 200)) {
+            errors.push(error);
+        }
+    }
+};
+
+export const validateAssociatedLink = (associatedLink: string, errors: ErrorInfo[], errorId: string) => {
+    try {
+        // eslint-disable-next-line @typescript-eslint/no-unused-vars, @typescript-eslint/ban-ts-comment
+        // @ts-expect-error
+        const url = new URL(associatedLink);
```

Comment:
This could just be `new URL(associatedLink);` without assigning it to a variable then we shouldn't need the eslint and ts disables

Reply:
done